Upgrade 9.0 updates (#800)
elastic/docs-content: deploy-manage/upgrade/deployment-or-cluster/kibana-roll-back.md

> # Roll back to a previous version of {{kib}} [upgrade-migrations-rolling-back]

Reviewer: Totally optional, just keeps the left nav title a bit shorter, so it doesn't span three lines.

```suggestion
---
navigation_title: "Roll back to a previous version"
---

# Roll back to a previous version of {{kib}} [upgrade-migrations-rolling-back]
```

> If you’ve followed [preparing for migration](/deploy-manage/upgrade/deployment-or-cluster/kibana.md#preventing-migration-failures) and [resolving migration failures](../../../troubleshoot/kibana/migration-failures.md), and {{kib}} is still unable to successfully upgrade, rollback {{kib}} until you’re able to identify and fix the root cause.

Reviewer: ```suggestion
If you’ve followed [preparing for migration](/deploy-manage/upgrade/deployment-or-cluster/kibana.md#preventing-migration-failures) and [resolving migration failures](../../../troubleshoot/kibana/migration-failures.md), and {{kib}} is still unable to successfully upgrade, roll back {{kib}} until you identify and fix the root cause.
```

::::{warning} 

> Before you roll back {{kib}}, ensure that the version you want to roll back to is compatible with your {{es}} cluster. If the version you want to roll back to is not compatible, you must also rollback {{es}}. Any changes made after an upgrade are lost when you roll back to a previous version.

Reviewer: ```suggestion
Before you roll back {{kib}}, ensure that the version you want to roll back to is compatible with your {{es}} cluster. If the version you want to roll back to is not compatible, you must also roll back {{es}}. Any changes made after an upgrade are lost when you roll back to a previous version.
```

::::

> To roll back after a failed upgrade migration, you must also roll back the {{kib}} feature state to be compatible with the previous {{kib}} version.

Reviewer: A couple of Qs:
- Is "upgrade migration" a term that we use elsewhere in the upgrade guide? IIRC, we typically only say "upgrade".
- "...you must also roll back the {{kib}} feature state..." – I'm a bit confused by the "also" here. Are rolling back Kibana (mentioned on line 3) and rolling back the Kibana feature state two different processes? Saying "also" implies that you should first roll back Kibana and then (also) roll back the Kibana feature state. But on this page we're only describing rolling back the Kibana feature state, which makes it seem like that's the only process involved.

Author: Good point for the first item. For the second item, I'll omit that sentence for now.

## Roll back by restoring the {{kib}} feature state from a snapshot [_roll_back_by_restoring_the_kib_feature_state_from_a_snapshot] 

1. Before proceeding, [take a snapshot](../../tools/snapshot-and-restore/create-snapshots.md) that contains the `kibana` feature state. By default, snapshots include the `kibana` feature state.

> 2. To make sure no {{kib}} instances are performing an upgrade migration, shut down all {{kib}} instances.

Reviewer: Same question about the "upgrade migration" term.

> 3. [Restore](../../tools/snapshot-and-restore/restore-snapshot.md) the `kibana` feature state from a snapshot taken before the failed {{kib}} upgrade. The following {{es}} request will only restore the {{kib}} feature state

Reviewer: ```suggestion
3. [Restore](../../tools/snapshot-and-restore/restore-snapshot.md) the `kibana` feature state from a snapshot taken before the failed {{kib}} upgrade. The following {{es}} request will only restore the {{kib}} feature state:
```

    ```console
    POST _snapshot/my_repository/my_snapshot_2099.05.06/_restore
    {
      "indices": "-*", <1>
      "feature_states": ["kibana"]
    }
    ```

>     1. Exclude all indices and data streams from the restore operation to ensure that only the {{kib}} system indices included in the {{kib}} feature state will be restored.

Reviewer: ```suggestion
    1. Exclude all indices and data streams from the restore operation to ensure that only the {{kib}} system indices included in the {{kib}} feature state are restored.
```

> 4. Start all {{kib}} instances on the older version you want to rollback to.

Reviewer: ```suggestion
4. Start all {{kib}} instances on the older version you want to roll back to.
```
